server/mongo_data_access.go

```diff
@@ -13,12 +13,61 @@ import (
 )
 
 // NewMongoDataAccessLayer returns an implementation of DataAccessLayer that is backed by a Mongo database
-func NewMongoDataAccessLayer(db *mgo.Database) DataAccessLayer {
-	return &mongoDataAccessLayer{Database: db}
+func NewMongoDataAccessLayer(db *mgo.Database, interceptors map[string]InterceptorList) DataAccessLayer {
+	return &mongoDataAccessLayer{
+		Database:     db,
+		Interceptors: interceptors,
+	}
 }
 
 type mongoDataAccessLayer struct {
-	Database *mgo.Database
+	Database     *mgo.Database
+	Interceptors map[string]InterceptorList
+}
+
+// An interceptor executes a function on a specified resource type immediately BEFORE
```

Comment:
Yeah, I remember reading that somewhere in the Golang spec. Thanks for the reminder!